src/mechanics/mesh/MeshFemDofConvert.cpp

```diff
@@ -1,96 +1,176 @@
+    return v3d;
+}
+
+struct NodePoint : Eigen::Vector3d // to inherit operator[]
 {
-    Eigen::VectorXd operator()(const DummyNode& node) const
+    NodePoint(Eigen::VectorXd v, NuTo::NodeSimple& node)
+        : Eigen::Vector3d(To3D(v))
+        , mNode(node)
     {
-        return node.coords;
     }
+    NuTo::NodeSimple& mNode;
 };
 
-
-// create a vector containing all coordinates of the new nodes. Contains a lot of duplicate dummy nodes.
-std::vector<DummyNode> GetNewDummyNodes(const NuTo::MeshFem& mesh, const NuTo::InterpolationSimple& interpolation)
+SubBoxes<NodePoint>::Domain SetupSubBoxDomain(const NuTo::MeshFem& mesh, int numBoxesPerDirection, double eps)
 {
-    std::vector<DummyNode> dummyNodes;
+    Eigen::VectorXd start = mesh.Elements.front().CoordinateElement().GetNode(0).GetValues();
+    Eigen::VectorXd end = start;
+
+    unsigned long numNodesTotal = 0;
     for (auto& elementCollection : mesh.Elements)
     {
-        const auto& coordinateElement = elementCollection.CoordinateElement();
-        for (int iNode = 0; iNode < interpolation.GetNumNodes(); ++iNode)
+        const int numNodes = elementCollection.CoordinateElement().GetNumNodes();
+        numNodesTotal += numNodes;
+
+        for (int iNode = 0; iNode < numNodes; ++iNode)
         {
-            DummyNode n;
-            n.coords = Interpolate(coordinateElement, interpolation.GetLocalCoords(iNode));
-            dummyNodes.push_back(n);
+            Eigen::VectorXd coords = elementCollection.CoordinateElement().GetNode(iNode).GetValues();
+            start = start.array().min(coords.array()); // array view (.array()) allows coeffwise operations
+            end = end.array().max(coords.array());
         }
     }
-    return dummyNodes;
+    const int dimension = start.rows();
+    
+    SubBoxes<NodePoint>::Domain d;
+    d.start = To3D(start) - Eigen::Vector3d::Constant(eps);
+    d.end = To3D(end) + Eigen::Vector3d::Constant(eps);
+
+    d.subs = Eigen::Vector3i::Constant(1);
+
+    for (int i = 0; i < dimension; ++i)
+        d.subs[i] = numBoxesPerDirection;
+
+    d.searchRadiusSquared = eps * eps;
+    return d;
 }
 
+
 void NuTo::AddDofInterpolation(MeshFem* rMesh, DofType dofType, const InterpolationSimple& interpolation)
 {
-    // We will build a spatial container containing many duplicate DummyNodes at a
-    // given coordinate. 
-    auto dummyNodes = GetNewDummyNodes(*rMesh, interpolation);
-    NuTo::SpatialContainer<DummyNode, DummyNodeCoordinate> dummyNodesSpatial(dummyNodes);
-
-    // Plan:
-    // If those dummynodes have _NO_ existingNodePtr, a new node
-    // is added to the rMesh->Nodes container. A reference to rMesh newly created node
-    // is saved at all of those duplicate dummy nodes.
+    // Setup subbox. These argument values are quite arbibrary and should maybe be chosen based on 
+    // the dimensions of the mesh.
+    SubBoxes<NodePoint> subBoxes(SetupSubBoxDomain(*rMesh, /*numBoxesPerDirection=*/300, /*eps=*/1.e-10));
```

Comment:
Maybe we should open a new issue to benchmark this and find a mapping between the mesh & element size to these values?